host_core/lib/host_core/linkdefs/manager.ex

```diff
@@ -62,7 +66,8 @@ defmodule HostCore.Linkdefs.Manager do
     :ets.insert(table_atom(lattice_prefix), {ld.id, ld})
   end
 
-  def uncache_link_definition(lattice_prefix, ldid) do
+  def uncache_link_definition(lattice_prefix, ldid, js_domain) do
+    HostCore.Jetstream.Client.kv_del(lattice_prefix, js_domain, "LINKDEF_#{ldid}")
```

Comment:
This doesn't feel right, based off the name of this function. It feels like both `uncache_link_definition` and `kv_del` should be called together, in most cases, but removing from an in-memory cache and removing from the source of truth feel like two different operations

Reply:
I'm curious about these two flows now:

With control interface client prior to v0.23:
- `wash ctl link del....`
- request issued to `wasmbus.ctl.[lattice].linkdefs.del`
- handled by `del_link_definition_by_triple`
- `del_link_definition` called
- `kv_del` called, which publishes to KV topic
- triggers a `handle_action` in `MetadataCacheLoader` (?)
- `del_link_definition` called (again)
- `kv_del` called (again), which publishes to KV topic again (no-op?)

With control interface client v0.23:
- `wash ctl link del...`
- deletes from KV directly
- triggers a `handle_action` in `MetadataCacheLoader`
- `del_link_definition` called
- `kv_del` called, which publishes to KV topic, but the key is already deleted, so no-op

Assuming I've followed the flow correctly, neither of these feel quite right. I think the flows should be:

Responding to a request on `request to `wasmbus.ctl.[lattice].linkdefs.del` (which is only hit by deprecated clients):
- simply call `kv_del`, deleting the key from the bucket
- (the `MetadataCacheLoader` should then get and handle a `:key_deleted` action, as below)

Responding to `:key_deleted` actions:
- uncache link def from ETS table
- broadcast event to other hosts in lattice

Reply:
The motivation behind deleting from the in-memory cache and calling the KV del is to handle the off chance that the key deleted message doesn't make its way to the host.

Reply:
> (the MetadataCacheLoader should then get and handle a :key_deleted action, as below)
This isn't 100% guaranteed. I'd rather delete it immediately especially since there's really no downside.

Reply:
>broadcast event to other hosts in lattice
This shouldn't be done because all other hosts in the lattice will have already received the key deleted message from the bucket subscription.